benchmarking/simulation/simulation.py

```diff
@@ -67,4 +76,9 @@ def run(self, num_runs: int) -> SimulationArtifact:
             self.run_times.append(end_time - start_time)
             self.team_sets.append(team_set)
 
+            # Save result to cache. Do this inside the loop so that if the program crashes, we still have the results from the previous runs.
+            if self.settings.cache_key:
```

Comment:
probably include a stagger to this -> grab the counter for the loop (I guess give it a name now) and do a `if i % SimulationCache.CACHE_EVERY_N_TRIALS` and add this constant to SimulationCache 

Kinda just spitballing though, put the constant wherever makes the most sense and add the logic where makes the most sense too.

More complex idea is to have a method in SimulationCache that's like `add_record(self, team_set: TeamSet, run_time: float)` and then inside that method, it saves to a dict it's keeping track of internally and chooses to do an append to the file when it decides it's worth doing, but this is kinda more thoughts and something to consider after perhaps

Reply:
Added the add_run() method, but not adding the `CACHE_EVERY_N_TRIALS` as performance is not a worry here